File: iOS8_Primer_ObjC/PlaceMap/PlaceMap/MapDisplayViewController.m
//
//  MapDisplayViewController.m
//  PlaceMap
//
//  Created by demo on 2014/09/14.
//  Copyright (c) 2014年 msyk.net. All rights reserved.
//

#import "MapDisplayViewController.h"
#import "AppDelegate.h"
#import "PlaceDatabase.h"
#import <MapKit/MapKit.h>

@interface MapDisplayViewController ()

@property (weak, nonatomic) IBOutlet MKMapView *mapView;
@property (weak, nonatomic) IBOutlet UILabel *placeName;

@end

@implementation MapDisplayViewController

- (void)viewDidLoad {
    NSLog(@"%s", __FUNCTION__);
    [super viewDidLoad];
    
    if([UIDevice currentDevice].userInterfaceIdiom
       == UIUserInterfaceIdiomPad) {
        UISplitViewController *svc = self.splitViewController;
        UINavigationItem *navItem = self.navigationItem;
        navItem.leftBarButtonItem = svc.displayModeButtonItem;
 //       navItem.leftItemsSupplementBackButton = YES;
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewDidAppear:(BOOL)animated
{
    NSLog(@"%s", __FUNCTION__);
    NSLog(@"%@", self.navigationItem);
    NSLog(@"%@", self.navigationController.navigationBar.items);
    
    [super viewDidAppear: animated];
    
    AppDelegate *appDelegate = [UIApplication sharedApplication].delegate;
    NSArray *places = appDelegate.placeDB.places;
    
    NSLog(@"Selected = %@", places[self.selectedIndex][@"name"]);
    
    self.placeName.text = [NSString stringWithFormat: @"%@ [%@]",
                           places[self.selectedIndex][@"pref"],
                           places[self.selectedIndex][@"name"]];
    NSNumber *latitude = places[self.selectedIndex][@"latitude"];
    NSNumber *longitude = places[self.selectedIndex][@"longitude"];
    CLLocationCoordinate2D coordinate = {
        .latitude = latitude.doubleValue,
        .longitude = longitude.doubleValue
    };
    MKCoordinateSpan span = {
        .latitudeDelta = 0.01,
        .longitudeDelta = 0.01
    };
    MKCoordinateRegion region = {
        .center = coordinate,
        .span = span
    };
    self.mapView.region = region;
}

@end

File: iOS9_Adv_Training/PlaceMap_Test/PlaceMap/MapDisplayViewController.swift
//
//  MapDisplayViewController.swift
//  PlaceMap
//
//  Created by Masayuki Nii on 2015/10/04.
//  Copyright © 2015年 Masayuki Nii. All rights reserved.
//

import UIKit
import MapKit

class MapDisplayViewController: UIViewController {
    
    var selectedIndex : Int = 0
    
    @IBOutlet weak var mapView: MKMapView!
    @IBOutlet weak var placeName: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if (UIDevice.currentDevice().userInterfaceIdiom == .Pad) {
            let svc = self.splitViewController
            let navItem = self.navigationItem
            navItem.leftBarButtonItem = svc?.displayModeButtonItem()
        }
        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false
        
        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
        // self.navigationItem.rightBarButtonItem = self.editButtonItem()
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        print("Selected = \(self.selectedIndex)", terminator: "")
        
        let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
        let places = appDelegate.placeDB.places
        
        let prefName = places[selectedIndex]["pref"] as! String
        let capName = places[selectedIndex]["name"] as! String
        print("Capital City = \(capName)")
        
        self.placeName.text = "\(prefName) [\(capName)]"
        
        let coordinate = CLLocationCoordinate2D(
            latitude: places[selectedIndex]["latitude"] as! CLLocationDegrees,
            longitude: places[selectedIndex]["longitude"] as! CLLocationDegrees
        )
        let span = MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)
        let region = MKCoordinateRegion(center: coordinate, span: span)
        self.mapView.region = region
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
}
